sycl/source/detail/device_info.hpp

```diff
@@ -301,6 +301,25 @@ struct get_device_info_impl<std::vector<memory_scope>,
   }
 };
 
+// Specialization for cuda cluster group
+template <>
+struct get_device_info_impl<bool, info::device::ext_oneapi_cuda_cluster_group> {
+  static bool get(const DeviceImplPtr &Dev) {
+    bool result = false;
+    if (Dev->getBackend() == backend::ext_oneapi_cuda) {
+      sycl::detail::pi::PiResult Err =
+          Dev->getPlugin()->call_nocheck<PiApiKind::piDeviceGetInfo>(
+              Dev->getHandleRef(),
+              PiInfoCode<info::device::ext_oneapi_cuda_cluster_group>::value,
+              sizeof(result), &result, nullptr);
+      if (Err != PI_SUCCESS) {
+        return false;
+      }
+    }
```

Comment:
In principle I would agree: the of `ext_oneapi_cuda_cluster_group` could in theory return true for backends other than cuda in the future.

In practice whilst other vendors may come up with similar features and we may be able to partially amalgamate them into a single aspect I doubt very much that we can avoid having `ext_oneapi_cuda_cluster_group` to represent the cuda specific cluster_group, that has various features that are extremely specific to the particular way that NVIDIA has chosen to accelerated cross work group collectives (in particular GEMM).

Since no other backend is ever expected to support this aspect, I think it is more efficient for the implementation to return false directly in the sycl runtime for this aspect.

We could do as you suggest, and I don't feel very strongly on this point, but with the impending removal of PI, I recommend that we do it in a follow up after PI is removed.

Reply:
The performance is a good point, but on the other hand it is redundant work when the backend is CUDA, so it really depend on what the hot path is expected to be. The additional overhead should be fairly cheap in either solution though, so I am fine with either, but personally I prefer to have the runtime be as ignorant to the backend as possible.

Reply:
Perhaps I misunderstand you but for the cuda backend it isn't actually a redundant call: this features is only supported on sm_90 and later.

Reply:
> Perhaps I misunderstand you but for the cuda backend it isn't actually a redundant call: this features is only supported on sm_90 and later.

What I mean is, we have two options: First is to check for the backend and fail fast or don't check the backend and fail slow(er).

For the fail-fast solution, when doing it on a CUDA device it would be faster to not check the backend and go directly to checking if the feature is supported for the device (sm_90 or later.) It is really negligible, but if we expect that users will mainly use this on CUDA devices, I would argue that paying a little extra when the backend is not CUDA should be fine.

Reply:
Sure, it's fine to have it a follow-up. It's not a big problem.

Reply:
> > Perhaps I misunderstand you but for the cuda backend it isn't actually a redundant call: this features is only supported on sm_90 and later.
> 
> What I mean is, we have two options: First is to check for the backend and fail fast or don't check the backend and fail slow(er).
> 
> For the fail-fast solution, when doing it on a CUDA device it would be faster to not check the backend and go directly to checking if the feature is supported for the device (sm_90 or later.) It is really negligible, but if we expect that users will mainly use this on CUDA devices, I would argue that paying a little extra when the backend is not CUDA should be fine.

I see what you mean thanks, it is a valid point.